crypto/bio/b_addr.c

```diff
@@ -565,7 +566,8 @@ static int addrinfo_wrap(int family, int socktype,
                          unsigned short port,
                          BIO_ADDRINFO **bai)
 {
-    OPENSSL_assert(bai != NULL);
+    if (bai == NULL)
```

Comment:
You can only safely remove a NULL check if you can ensure that no code path relies on that behaviour.